src/network/mtp/threads.cpp

```diff
@@ -317,6 +317,7 @@ void ConnectionSendThread::sendAsPacketReliable(BufferedPacketPtr &p, Channel *c
 		channel->outgoing_reliables_sent.insert(p,
 			(channel->readOutgoingSequenceNumber() - MAX_RELIABLE_WINDOW_SIZE)
 				% (MAX_RELIABLE_WINDOW_SIZE + 1));
+		// wtf is this calculation?? ^
```

Comment:
start of the reliable window for that packet?

Reply:
can you explain how we arrive at this number?
because this calculation falls apart with max=2048

Reply:
The whole calculation I think is pointless, since this only exists cause someone had the idea that incoming and outgoing packets should be tracked using the same queue type despite clearly having different requirements.

I *think* the idea was to get the actual current sequence number, subtract from it the window size, and then take that mod the size of the sequence number type. Someone messed up and used `MAX_RELIABLE_WINDOW_SIZE + 1` instead of  `MAX_SEQ_NUM` or whatever the (stupid-ish) `#define` for that is.
Didn't matter before your changes since it was basically pointless code (I think) that was just meant to pass a window check and it passed.

I made changes to stop using that common queue type on the receiving end when implementing encryption because it involved pointless constructing a wire formatted packet, maybe I should pull those changes out and also refactor the sender side to use a more sane queue.

Reply:
> The whole calculation I think is pointless, since this only exists cause someone had the idea that incoming and outgoing packets should be tracked using the same queue type despite clearly having different requirements.

> Didn't matter before your changes since it was basically pointless code (I think) that was just meant to pass a window check and it passed.

that's what I thought too, however that can't be entirely true as depending on the current connection state - out of of two outgoing reliables you want to track with seqnums 1234, 65500 - either 65500 is the first or 1234 is the first and you need some information to disambiguate.

anyway it doesn't really matter. personally I wouldn't invest more work into the protocol code beyond fixing the necessary.